packages/encoding/src/decoding/decodeDexOutput.ts

```diff
@@ -0,0 +1,50 @@
+import { PedersenHash } from '@explorer/types'
+
+import { StarkExDexOutput } from '../OnChainData'
+import { ByteReader } from './ByteReader'
+
+export function decodeDexOutput(data: string): StarkExDexOutput {
+  const reader = new ByteReader(data)
+
+  const configCode = reader.readBigInt(32)
+  const initialValidiumVaultRoot = PedersenHash(reader.readHex(32))
+  const finalValidiumVaultRoot = PedersenHash(reader.readHex(32))
+  const initialRollupVaultRoot = PedersenHash(reader.readHex(32))
+  const finalRollupVaultRoot = PedersenHash(reader.readHex(32))
+  const initialOrderRoot = PedersenHash(reader.readHex(32))
+  const finalOrderRoot = PedersenHash(reader.readHex(32))
+  const globalExpirationTimestamp = reader.readNumber(32)
+  const validiumVaultTreeHeight = reader.readNumber(32)
+  const rollupVaultTreeHeight = reader.readNumber(32)
+  const orderTreeHeight = reader.readNumber(32)
+  const modificationsCount = reader.readNumber(32)
+  const conditionalTransfersCount = reader.readNumber(32)
+  const L1VaultUpdatesCount = reader.readNumber(32)
+  const L1OrderMessagesCount = reader.readNumber(32)
+
+  if (!reader.isAtEnd()) {
+    // https://github.com/starkware-libs/starkex-contracts/blob/75c3a2a8dfff70604d851fc6b1a2bc8bc1a3964b/scalable-dex/contracts/src/components/OnchainDataFactTreeEncoder.sol#L12
+    // When reading calldata from updateState two new values are appended
+    reader.skip(64)
```

Comment:
I think we should remove this whole `.skip()` conditional block. If I remember correctly, it was added because ProgramOutput on Validium has 2 more fields that we didn't understand, so we decided to temporarily skip them. But here we assume we know the whole structure so we can `.assertEnd()`.

Reply:
Done